File: contracts/responses/SearchResponses.ts
import { SearchResult } from "../models/SearchResult";
import { PageInfo } from "./Pagination";

export interface SearchFacet {
  category: string;
  value: string;
  count: number;
}

export interface SearchResponse {
  query: string;
  results: SearchResult[];
  facets: SearchFacet[];
  pageInfo: PageInfo;
}

File: contracts/responses/SearchResponses.kt
package contracts.responses

import contracts.models.SearchResult
import kotlinx.serialization.Serializable

@Serializable
data class SearchFacet(
    val category: String,
    val value: String,
    val count: Long
)

@Serializable
data class SearchResponse(
    val query: String,
    val results: List<SearchResult> = emptyList(),
    val facets: List<SearchFacet> = emptyList(),
    val pageInfo: PageInfo
)
